src/app/component/user/components/profile/profile-cards/profile-cards.component.ts

```diff
@@ -1,23 +1,39 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { catchError } from 'rxjs/operators';
+import { Subscription } from 'rxjs';
+import { CardModel } from '@user-models/card.model';
 import { ProfileService } from '../profile-service/profile.service';
 
 @Component({
   selector: 'app-profile-cards',
   templateUrl: './profile-cards.component.html',
   styleUrls: ['./profile-cards.component.scss']
 })
-export class ProfileCardsComponent implements OnInit {
+export class ProfileCardsComponent implements OnInit, OnDestroy {
+  public profileSubscription: Subscription;
+  public factOfTheDay: CardModel;
+  public error;
 
-   constructor(private profileService: ProfileService) { }
-
-   public profileSubscription;
-   public facts = [];
+  constructor(private profileService: ProfileService) { }
 
   ngOnInit() {
-    this.profileSubscription = this.profileService.getFactsOfTheDay().subscribe(
-      (success) => {
-        this.facts = [success, ...this.facts];
-      }
-    );
+    this.getFactOfTheDay();
+  }
+
+  getFactOfTheDay(): void {
+    this.profileSubscription = this.profileService.getFactsOfTheDay()
+      .pipe(
+        catchError((error) => {
+          this.error = error;
+          return error;
```

Comment:
Why you return an error? 
How you can understand that error happened ?

Reply:
The catchError operator is going to take the error and pass it to the error handling function.
That function is expected to return an Observable which is going to be a replacement Observable for the stream that just errored out.

If error happened, the component would not appear on the screen